cmd/tsidp/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# Build stage
+FROM golang:alpine AS builder
+
+# Install build dependencies
+RUN apk add --no-cache git
+
+# Set working directory
+WORKDIR /src
+
+# Copy only go.mod and go.sum first to leverage Docker caching
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the entire repository
+COPY . .
+
+# Build the tsidp binary
+RUN CGO_ENABLED=0 go build -o /bin/tsidp ./cmd/tsidp
+
+# Final stage
+FROM alpine:latest
+
+# Install required dependencies
+RUN apk add --no-cache ca-certificates
```

Comment:
We embed LetsEncrypt already so I don't think this is needed. This is fine to keep but maybe acknowledge that in a comment at least that we know that we don't strictly need these.